### PR TITLE
feat: auto-updating mocks

### PR DESCRIPTION
**Pull request by goosewobbler**
TODO:

- [x] Extract mock storage
- [x] Fix `mockAll` case
- [ ] Fix `mockImplementation` case
- [ ] Update mock interface

### CODE REVIEW

File: example-cjs/e2e/api.spec.ts [hunk 1]
```diff
@@ -19,9 +19,9 @@ describe('mocking', () => {
           }),
       );
 
-      const mockedShowOpenDialog = await showOpenDialog.update();
-      expect(mockedShowOpenDialog).toHaveBeenCalledTimes(1);
-      expect(mockedShowOpenDialog).toHaveBeenCalledWith({
+      // const mockedShowOpenDialog = await showOpenDialog.update();
```
Comment (christian-bromann) [inline example-cjs/e2e/api.spec.ts:22]:
```suggestion
```

File: example/e2e/api.spec.ts [hunk 2]
```diff
@@ -19,9 +19,9 @@ describe('mocking', () => {
           }),
       );
 
-      const mockedShowOpenDialog = await showOpenDialog.update();
-      expect(mockedShowOpenDialog).toHaveBeenCalledTimes(1);
-      expect(mockedShowOpenDialog).toHaveBeenCalledWith({
+      // const mockedShowOpenDialog = await showOpenDialog.update();
```
Comment (christian-bromann) [inline example/e2e/api.spec.ts:22]:
```suggestion
```

File: src/service.ts [hunk 3]
```diff
@@ -60,4 +60,22 @@ export default class ElectronWorkerService implements Services.ServiceInstance {
       }
     }
   }
+
+  async afterCommand(commandName: string) {
+    if (commandName === 'execute') {
```
Comment (christian-bromann) [inline src/service.ts:65]: Why do we run this only after `execute` commands?

Comment (christian-bromann) [reply src/service.ts:65]: Also I wonder how feasible it is to check if any mocks have been registered before and skip this if there aren't any.

Comment (goosewobbler) [reply src/service.ts:65]: Perhaps we can store a value of the number of mocks registered and remove the `execute` check.  This PR is just a rough prototype atm, the mock interface needs updating for the new approach amongst other things.

Comment (christian-bromann) [reply src/service.ts:65]: Sounds good. I would just put the mock store into a file and have it exported only a single instance. This way you can import it from everywhere to access its values, kind a like a global

**Comment by goosewobbler** [comment]
Closing in favour of #346

**Pull request closed without merging**